lib/sudo-pam/src/converse.rs

```diff
@@ -136,17 +139,17 @@ where
 pub struct CLIConverser;
 
 impl SequentialConverser for CLIConverser {
-    fn handle_normal_prompt(&self, msg: &str) -> PamResult<String> {
+    fn handle_normal_prompt(&self, msg: &str) -> PamResult<RawBytes> {
         print!("[Sudo: input needed] {msg}");
         std::io::stdout().flush().unwrap();
 
-        let mut s = String::new();
-        std::io::stdin().lock().read_line(&mut s).unwrap();
+        let mut s = Vec::new();
+        std::io::stdin().lock().read_until(0x0a, &mut s).unwrap();
```

Comment:
Maybe we add a little const here to get rid of the magic `0x0a`.

Reply:
Also: this may re-allocate the vec while reading the password, and that memory will not be cleared.

Reply:
Note, most of the comments that follow relate to the original `rpassword` (this code as a minimal adaptation of the original); i.e. what we want is actually a "better rpassword".

This last point is something that's a good observation (and occurs in `rpassword`), we should maybe submit a patch upstream.

Reply:
So *if* we are going to the trouble to zero out memory, we should also not use BufRead for the reason you give (data will be stored in a buffer). In fact we could even go as far as use the file descriptor to read data from the tty using the posix interface.

(Of course the only point in taking care to wipe memory is to make sure it does't accidentally appear somewhere as uninitialized memory.)